lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -1201,6 +1223,10 @@ void ProxyRenderDelegate::_Execute(const MHWRender::MFrameContext& frameContext)
                 HdPrimTypeTokens->material, SdfPath::AbsoluteRootPath());
             for (auto material : materials) {
                 changeTracker.MarkSprimDirty(material, HdMaterial::DirtyParams);
+                // Tell all the Rprims associated with this material to recompute primvars
+                HdVP2Material* vp2material = static_cast<HdVP2Material*>(
+                    _renderIndex->GetSprim(HdPrimTypeTokens->material, material));
+                vp2material->MaterialChanged(_sceneDelegate.get());
```

Comment:
Why MarkSprimDirty is not enough here? Shouldn't we fix the Sync function instead?

Reply:
When we are in the render delegate Sync function of the material, it is already too late since the scene delegate updates are already completed and the render delegate has no API to dirty the scene delegate to add the newly discovered primvar. 

This is why I need to mark the Geoms dirty to allow a primvar query to be run in the scene delegate to add any missing UV streams.